src/store/modules/DownloadModule.ts

```diff
@@ -125,28 +127,67 @@ export const DownloadModule = {
             });
         },
 
-        async downloadAndInstallSpecific({state, commit, dispatch}, params: {
-            combo: ThunderstoreCombo,
-            profile: ImmutableProfile
+        async download({state, commit, dispatch}, params: {
+            combos: ThunderstoreCombo[],
+            downloadId: UUID
         }) {
-            const downloadId = await dispatch('addDownload', [`${params.combo.getMod().getName()} (${params.combo.getVersion().getVersionNumber().toString()})`]);
-
             try {
-                const downloadedMods = await ThunderstoreDownloaderProvider.instance.download(
-                    params.profile,
-                    params.combo,
+                await ThunderstoreDownloaderProvider.instance.downloadCombos(
+                    params.combos,
                     state.ignoreCache,
-                    (downloadProgress: number, modName: string, status: number, err: R2Error | null) => {
-                        dispatch('downloadProgressCallback', { downloadId, downloadProgress, modName, status, err });
+                    (downloadProgress, modName, status, err) => {
+                        dispatch('downloadProgressCallback', { downloadId: params.downloadId, downloadProgress, modName, status, err });
                     }
                 );
-                await dispatch('installMods', {downloadedMods, downloadId, profile: params.profile});
             } catch (e) {
-                commit('updateDownload', { downloadId, failed: true });
-                throw e;
+                commit('updateDownload', { downloadId: params.downloadId, failed: true });
+                commit('error/handleError', R2Error.fromThrownValue(e), { root: true });
             }
         },
 
+        downloadAndInstallCombos({commit, dispatch}, params: {
+            combos: ThunderstoreCombo[],
+            profile: ImmutableProfile,
+            downloadId: UUID
+        }) {
+            setTimeout(async () => {
```

Comment:
Nitpick: doing `const {combos, profile, downloadId} = params;` would somewhat avoid repetition.

Also I think I've asked this before in some context but as the location of the code has changed again: is the setTimeout still necessary?

Reply:
I removed the `setTimeout` and used that params deconstructing in a few places.